packages/loader/container-loader/src/container.ts

```diff
@@ -1305,7 +1306,8 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     }
 
     private async getDocumentStorageService(): Promise<IDocumentStorageService> {
-        return this._deltaManager.connectToStorage();
+        const storage = await this._deltaManager.connectToStorage();
+        return new ProtocolTreeStorageService(storage, ()=>this.captureProtocolSummary());
```

Comment:
this pattern is a bit odd, but i prefered it over passing all the necessary parts into the runtime to generate it there. additionally, the loader is host code, and could protentially be across a security boundry, like and iframe, in which case we could keep the protocol tree generation more secure than it would be if the runtime own this.